### PR TITLE
Update for active voice, clarity and grammar

### PR DESCRIPTION
**Pull request by AnotherButler**
Compiler configuration section not included because of open PR #453

### CODE REVIEW

**Comment by screamerbg** [comment]
@AnotherButler could you please confirm when you're ready with the changes/fixes?

File: README.md [hunk 1]
```diff
@@ -661,123 +658,99 @@ my-mbed-os-example (a5ac4bf2e468)
...
+Git enables a workflow where the publish/push repository may be different than the original ("origin") one. This allows new revisions in a fork repository while maintaining an association with the original repository. To use this workflow, first import an mbed OS program or mbed OS itself, and then associate the push remote with your fork. For example:
 
 ```
 $ git remote set-url --push origin https://github.com/screamerbg/repo-fork
 ```
 
-Each time you `git` commit and push, or use `mbed publish`, the new revisions will be pushed against your fork. You can fetch from the original repository using `mbed update` or `git pull`. If you explicitly want to fetch or pull from your fork, then you can use `git pull https://github.com/screamerbg/repo-fork [branch]`.
+Each of `git commit`, `git push` and `mbed publish` push the new revisions to your fork. You can fetch from the original repository using `mbed update` or `git pull`. If you explicitly want to fetch or pull from your fork, then you can use `git pull https://github.com/screamerbg/repo-fork [branch]`.
```
Comment (bridadan) [inline README.md]: This isn't quite right. `git commit` and `git push` are the equivalent of `mbed publish`. So they can't be flattened to a list. The two `git` commands must remain paired.

So perhaps 

> Both `git commit & git push` and `mbed publish` push the new revisions...

The `&` is valid command syntax, I'm not using it as an English shortcut :)

File: README.md [hunk 2]
```diff
@@ -661,123 +658,99 @@ my-mbed-os-example (a5ac4bf2e468)
...
-The update command will fail if there are changes in your program or library that `update` could overwrite. This is by design: mbed CLI does not run operations that would result in overwriting local changes that are not yet committed. If you get an error, take care of your local changes (commit or use one of the options below), then rerun `update`.
+The update command fails if there are changes in your program or library that `mbed update` could overwrite. This is by design. mbed CLI does not run operations that would result in overwriting uncommitted local changes. If you get an error, take care of your local changes (commit or use one of the options below), and then rerun `mbed update`.
 
 ### Updating to an upstream version
 
-___Updating a program___
+#### Updating a program
 
-To update your program to another upstream version, go to the root folder of the program and run:
+To update your program to another upstream version, go to the root folder of the program, and run:
```
Comment (bridadan) [inline README.md]: Oxford comma? Or is this a corner case :smile: (I am actually curious though!)

Comment (AnotherButler) [inline README.md]: Haha, no. When a coordinating conjunction joins two independent clauses, you must add a comma before the conjunction.

**Comment by AnotherButler** [comment]
@screamerbg and @bridadan I think this is ready now. Please confirm.

**Comment by screamerbg** [comment]
Thanks @AnotherButler

**Pull request merged**